allennlp/tests/data/dataset_readers/atis_test.py

```diff
@@ -0,0 +1,68 @@
+# pylint: disable=no-self-use,invalid-name
+from allennlp.data.dataset_readers import AtisDatasetReader
+from allennlp.common.testing import AllenNlpTestCase
+
+from allennlp.semparse.worlds import AtisWorld
+
+class TestAtisReader(AllenNlpTestCase):
+    def test_atis_read_from_file(self):
+        reader = AtisDatasetReader()
+        data_path = AllenNlpTestCase.FIXTURES_ROOT / "data" / "atis" / "sample.json"
+
+        instances = list(reader.read(str(data_path)))
+
+        assert len(instances) == 11
+        instance = instances[0]
+
+        assert instance.fields.keys() == \
+                {'utterance',
+                 'actions',
+                 'world',
+                 'target_action_sequence',
+                 'linking_scores'}
+
+        assert [t.text for t in instance.fields["utterance"].tokens] == \
+                ['show', 'me', 'the', 'one', 'way',
+                 'flights', 'from', 'detroit', 'to',
+                 'westchester', 'county']
+
+        assert isinstance(instance.fields['world'].as_tensor({}), AtisWorld)
+        # Check that the strings in the actions field has the same actions
+        # as the strings in the world.
+        valid_strs = set()
+        for prod in instance.fields['actions'].field_list:
+            if prod.rule.startswith('string'):
+                valid_strs.add(prod.rule)
+
+        world = instance.fields['world'].metadata
+        assert valid_strs == set(world.valid_actions['string'])
+
+        # We make sure all the entities are sorted in reverse order because
+        # we need to enforce some ordering for it to match the correct row
+        # in the linking scores.
+        assert sorted(world.valid_actions['string'], reverse=True) == \
```

Comment:
Instead of testing this, just have a test that explicitly tests the linking scores (you didn't actually test the values of it below).  If you're testing values, and have [comments on each line saying which entity is being testing](https://github.com/allenai/allennlp/blob/4fa4dc23ccc5a0f5332d73e8da7688d343296f82/allennlp/tests/data/fields/knowledge_graph_field_test.py#L120-L123), you'll be testing that the alignment is correct.  You don't need to test a particular arbitrary ordering here, just that the linking scores come out right.  Does this make sense?